src/haddock/gear/prepare_run.py

```diff
@@ -106,52 +115,100 @@ def setup_run(workflow_path, restart_from=None):
         Whether to erase the previous run folder and reprare from
         scratch. Defaults to `True`.
 
+    start_from_copy : str or Path
+        The path created with `haddock3-copy` to start the run from.
+        Defaults to None, do not use this option.
+
     Returns
     -------
     tuple of two dicts
         A dictionary with the parameters for the haddock3 modules.
         A dictionary with the general run parameters.
     """
+    from_scratch = restart_from is None and start_from_copy is None
+
     # read config
     params = read_config(workflow_path)
 
-    check_mandatory_argments_are_present(params)
-    validate_module_names_are_not_mispelled(params)
-    check_specific_validations(params)
-
     # update default non-mandatory parameters with user params
     params = recursive_dict_update(
         non_mandatory_general_parameters_defaults,
         params)
 
-    clean_rundir_according_to_restart(params[RUNDIR], restart_from)
-
-    # copy molecules parameter to topology module
-    copy_molecules_to_topology(params)
-    if len(params["topoaa"]["molecules"]) > max_molecules_allowed:
-        raise ConfigurationError("Too many molecules defined, max is {max_molecules_allowed}.")  # noqa: E501
-
     # separate general from modules parameters
     _modules_keys = identify_modules(params)
     general_params = remove_dict_keys(params, _modules_keys)
     modules_params = remove_dict_keys(params, list(general_params.keys()))
-    zero_fill.read(modules_params)
 
-    # populate topology molecules
-    populate_topology_molecule_params(modules_params["topoaa"])
+    if not_none(start_from_copy):
+        with suppress(TypeError):
+            start_from_copy = Path(start_from_copy)
 
-    populate_mol_parameters(modules_params)
+        general_params[RUNDIR] = start_from_copy
 
-    # validations
-    validate_modules_params(modules_params)
+    validate_module_names_are_not_mispelled(modules_params)
```

Comment:
Unrelated: I still like this misspelling of the word `mispelled` (should be `misspelled`).

Reply:
It has a very clear origin in my brain:
`mis-` is used to assign a "bad", "wrong" behavior to words -> https://www.macmillandictionary.com/dictionary/british/mis

That's why I constantly type `mispelled`, from wrongly-spelled. Clearly, it is wrong. The correct is `misspelled`. What should it be? Missed the correct spelling? :-P Thanks again @douweschulte